mono/metadata/threadpool-worker-default.c

```diff
@@ -751,6 +745,24 @@ monitor_thread (gpointer unused)
 				break;
 			}
 
+
+			gboolean limit_worker_max_reached = FALSE;
+			ThreadPoolWorkerCounter counter = { 0 };
+			int new_thread_count = 0;
+
+			COUNTER_ATOMIC (counter, {
+				new_thread_count = thread_count (counter) + 1;
+				if (new_thread_count > worker.limit_worker_max) {
+					limit_worker_max_reached = TRUE;
+					break;
+				}
```

Comment:
Two things:
1. I don't get why this code moved from the outer loop to inside the `for (i = 0; i < 5; ++i)` loop. 
2. The corresponding code in CoreCLR does actually mutate the active `max_working` count so this also seems like a change to the algorithm
 https://github.com/dotnet/coreclr/blob/5f5f9606a6eef78256536f667c2df7b2495a8c3d/src/vm/win32threadpool.cpp#L4329-L4342

Reply:
I agree.
I also don’t understand.
I need to compare to CLR also.

One thing I missed, is that there is an assignment to max_workers.

Perhaps that can lower it.
I don’t know the math behind it.

I wanted to kinda more favor unpark over createthread, and not increment max_workers if merely unparking.

I haven’t run the repro.
Between repro & print debugging & CLR to compare with, shouldn’t be *too* difficult.